src/main/scala/nl.knaw.dans.easy.deposit/docs/dm/Spatial.scala

```diff
@@ -25,35 +25,43 @@ object Spatial {
   val RD_SRS_NAME = "http://www.opengis.net/def/crs/EPSG/0/28992"
 }
 
-trait SchemedSpatial extends OptionalValue {
-  val scheme: Option[String]
-
+trait SchemedSpatial extends OptionalValue with OptionalScheme {
   lazy val srsName: String = {
     scheme match {
       case Some("degrees") => Spatial.DEGREES_SRS_NAME
       case Some("RD") => Spatial.RD_SRS_NAME
       case Some(s) if s.trim.nonEmpty => s
-      case _ => null // will suppress the XML attribute
+      case _ => throw new IllegalArgumentException(s"scheme is mandatory for $errorDetails")
     }
   }
+
+  private def errorDetails = s"${ getClass.getSimpleName }: ${ toJson(this) }"
+
+  protected def allOrNone(msg: String, args: Option[String]*): Unit =
```

Comment:
please annotate this and all use cases with an `@throws` if you don't want to return a `Try`